### PR TITLE
Improve password management and code complexity

### PR DESCRIPTION
**Pull request by Aminelouati**

### CODE REVIEW

**Comment by Mykhailenko** [comment]
Why do you have `if (subFiles != null)` in several places in the class? Does not seem that FTPClient::listFiles can return null...

**Comment by Aminelouati** [comment]
This test is necessary to check whether the current directory is empty or not.

**Comment by Mykhailenko** [comment]
@Aminelouati could you please point out to java doc where it is written that null can be returned.

File: scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/FTPConnector.java [hunk 1]
```diff
@@ -68,15 +68,25 @@
...
-    private String ftpMode = null;
+    private String ftpMode;
 
-    private String ftpUsername = null;
+    private String ftpUsername;
 
-    private String ftpPassword = null;
+    private String ftpPassword;
+
+    private String ftpUrlKey;
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/FTPConnector.java]: MINOR Remove the "ftpUrlKey" field and declare it as a local variable in the relevant methods. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1450)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**